app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+package com.rockthevote.grommet.ui.registration
+
+import android.util.Base64
+import com.rockthevote.grommet.R
+import com.rockthevote.grommet.data.db.model.*
+import com.rockthevote.grommet.ui.registration.address.AddressData
+import com.rockthevote.grommet.ui.registration.name.PersonNameData
+import com.rockthevote.grommet.util.Dates
+import java.util.*
+
+/**
+ * Handles transformation of [registrationData]
+ */
+class RegistrationDataTransformer @Throws(InvalidRegistrationException::class) constructor(
+    private val registrationData: RegistrationData,
+    private val partnerInformation: PartnerInformation
+) {
+
+    init {
+        validateRegistrationData(registrationData)
+    }
+
+    // These have already be evaluated as non null in [validateRegistrationData]
+    private val newRegistrationData = registrationData.newRegistrantData!!
+    private val addressData = registrationData.addressData!!
+    private val additionalInfoData = registrationData.additionalInfoData!!
+    private val assistanceData = registrationData.assistanceData!!
+    private val reviewData = registrationData.reviewData!!
+
+    /**
+     * Transforms the [registrationData] and [partnerInformation] into a [RockyRequest]
+     *
+     * @throws [InvalidRegistrationException] if unable to validate [RegistrationData]
+     */
+    fun transform(
+        unknown: UnknownDataSource
+    ): RockyRequest {
+
+        val body = RockyRequestBody(
+            lang = reviewData.formLanguage,
+            phoneType = null, // TODO I can't find where/what is supposed to populate this
+            partnerId = partnerInformation.partnerId,
+            optInEmail = additionalInfoData.hasOptedIntoNewsUpdates,
+            optInSms = additionalInfoData.hasOptedIntoNewsCallAndText,
+            optInVolunteer = additionalInfoData.hasOptedForVolunteerText,
+            // TODO All of these data sources are unknown, I think they may be from partner information, but don't know for sure
+            partnerOptInSms = unknown.partnerOptInSms,
+            partnerOptInEmail = unknown.partnerOptInEmail,
+            partnerOptInVolunteer = unknown.partnerOptInVolunteer,
+            finishWithState = true, // TODO is this a default value?
```

Comment:
yes.

default values [here](https://github.com/TrustTheVote-
Project/Grommet/blob/6b0230af2e1f500db3943544cc64a177bd44d0ed/app/src/main/java/com/rockthevote/grommet/data/api/model/ApiRockyRequest.java#L68)

basically the same comment as above.

Reply:
Just ensuring we're on the page here, when I'm referring to a "default value" I mean that this is the only value that will be populated into this field, rather than a fallback value. Is that still correct?